flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -255,23 +258,52 @@
 
     private final StreamTaskAsyncExceptionHandler asyncExceptionHandler;
 
-    /**
-     * Flag to mark the task "in operation", in which case check needs to be initialized to true, so
-     * that early cancel() before invoke() behaves correctly.
-     */
-    private volatile boolean isRunning;
-
-    /** Flag to mark the task at restoring duration in {@link #restore()}. */
-    private volatile boolean isRestoring;
-
-    /** Flag to mark this task as canceled. */
-    private volatile boolean canceled;
+    /** Possible states of a Task. */
+    private static class TaskState {
```

Comment:
Maybe is it better to put the inner class at the bottom of this class?

Reply:
Ack. moved to the bottom of the class